Wrappers/Python/cil/optimisation/functions/Function.py

```diff
@@ -488,14 +489,15 @@ def gradient(self, x, out=None):
 
         Returns
         -------
-        DataContainer, the value of the gradient of the scaled function evaluated at :math:`x` or `None` if `out`.
+        DataContainer, the value of the gradient of the scaled function evaluated at :math:`x`. 
 
         """
         if out is None:
             return self.scalar * self.function.gradient(x)
         else:
             self.function.gradient(x, out=out)
             out *= self.scalar
+            return out 
```

Comment:
Surely these patterns should be:

```suggestion
        res = self.function.gradient(x, out=out)
        res *= self.scalar
        return res
```